### DISCUSSION TITLE
Production optimizations don't seem to work when using the Tailwind CLI

### DISCUSSION THREAD

**Discussion by gamecubate** [tailwindlabs/tailwindcss#4886] [Help]
Production optimizations don't seem to work when using the Tailwind CLI despite having followed the docs.

Can you spot a mistake, if any?

Process:
```
npx tailwindcss init
touch index.html
# edit index.html and tailwind.config.js (see below)
NODE_ENV=production npx tailwindcss -i ./tailwind.css -o ./dist/tailwind.css
```

Result:
`dist/tailwind.css` file is almost as large as the original (4Mb)

index.html:
```
<!doctype html>
<html>
<head>
  <meta charset="UTF-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <link href="./tailwind.css" rel="stylesheet">
</head>
<body>
  <div  class="container mx-auto">
    <h1 class="text-5xl">Hello</h1>
  </div>
</body>
</html>
```

tailwind.config.js:
```
module.exports = {
  purge: {
    enabled: true,
    content: ['./*.html', './**/*.html']
  },
  darkMode: false,
  theme: {
    extend: {},
  },
  variants: {
    extend: {},
  },
  plugins: [],
}
```

**Comment by gamecubate** [comment]
Found a workaround but not documented. Tried with nodejs versions 12.20.1 _and_ 15.5.1

This -- the adding of the build command -- works (optimizes the size of output file):
```
$ NODE_ENV=production npx tailwindcss build ./tailwind.css -o ./dist/tailwind.css -c tailwind.config.js
npx: installed 128 in 4.792s
[deprecation] Running tailwindcss without -i, please provide an input file.
```

While this doesn't (doesn't optimize anything):
```
$ NODE_ENV=production npx tailwindcss build -i ./tailwind.css -o ./dist/tailwind.css
```

And neither does this (the documented way):
```
$ NODE_ENV=production npx tailwindcss -i ./tailwind.css -o ./dist/tailwind.css
```

**Comment by adamwathan** [comment]
I tried exactly what you explained and it worked here:

<img width="624" alt="image" src="https://user-images.githubusercontent.com/4323180/124402238-fda12d80-dcfc-11eb-8c25-7a4077acc8d3.png">

The built file is 12kb. Any chance you can share a GitHub repo that reproduces the issue to make it easier to troubleshoot?

**Reply by gamecubate** [reply]
I'm not sure it will help but [here it is](https://github.com/gamecubate/tailwind-hello-1).

Meanwhile ...
<img width="862" alt="Screen Shot 2021-07-06 at 8 15 33 AM" src="https://user-images.githubusercontent.com/1483884/124598507-b321cd00-de32-11eb-850e-46cc6258e6af.png">

**Reply by adamwathan** [reply]
The problem is your `tailwind.css` file. It contains all of the compiled CSS instead of the Tailwind directives. That file should just be this:

```css
@tailwind base;
@tailwind components;
@tailwind utilities;
```

**Reply by gamecubate** [reply]
Thanks for this. It worked.

infile.css:
```
@tailwind base;
@tailwind components;
@tailwind utilities;
```

Watch  for changes:
```
$ npx tailwindcss -i infile.css -o outfile.css --watch
```

index.html:
```
<!doctype html>
<html>
<head>
  <meta charset="UTF-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <link href="./outfile.css" rel="stylesheet">
</head>
<body>
  <h1 class="font-bold text-5xl text-red-500">TAILWIND-CSS</h1>
</body>
</html>
```

Refactor classes as components:
```
// infile.css
@tailwind base;
@tailwind components;
@tailwind utilities;

@layer components {
  .my-title {
    @apply font-bold text-5xl text-red-500;
  }
}

// index.html
...
  <h1 class="my-title"> TAILWIND-CSS</h1>
...
```

When ready to deploy hence optimize:
```
NODE_ENV=production npx tailwindcss  -i infile.css -o outfile.css
```
